app/grandchallenge/cases/models.py

```diff
@@ -1056,8 +1096,8 @@ def handle_event(self, *, event):
             self.save()
 
     def handle_completed_job(self, *, event):
-        image_set = self.validate_image_set(event=event)
-        self.convert_image_set_to_internal(image_set=image_set)
+        image_set_id = self.validate_image_set(event=event)
```

Comment:
The validate_image_set function does quite a lot. Maybe the responsibilities should be split out a bit. I wouldn't have thought that a validate function would return anything - if we follow Djangos design then cleaning stuff just raises errors when there are problems.